### ISSUE TITLE
Plumes disappearing mid flight, NREs when selected in VAB

### ISSUE THREAD

**Issue by zorg2044** [sarbian/SmokeScreen#35]
Hi Sarbian,

Outlining an issue with the new smokescreen. 

1) NREs in the console log when any part with a realplume config is selected.

reproduction steps: Happens when any part with a real plume is selected.

Log where I booted game and selected a few different parts in VAB

https://www.dropbox.com/s/ugt4frcq8drolbc/output_log.txt?dl=0

2) Plume disappearing mid-flight.

reproduction step: uncertain. Happened once during a heavily modded game playthrough, plume showed up on part on one burn and disappeared on the next. One other user on the BDB forum thread mentioned a similar issue. Unfortunately did not manage to save the KSP log for this.

Attempted to reproduce in clean install. As above, NREs in the VAB when part selected but NREs disappeared when launched into flight. Plume did not disappear in a couple of attempts. Based on the report from the other user mentioned, this seems inconsistent. I will update if I can reproduce this.

Log from attempt to reproduce, includes a couple of test flights: https://www.dropbox.com/s/mjdfjhzumd8ofu5/output_log.txt?dl=0

Both logs are from KSP 1.7.1 installs, windows 10, Smokscreen 2.8.2 with only RealPlume and Tantares LV installed. 

Rolling back to v.2.8.0 eliminates the NREs in the VAB.

I hope this is helpful! The new features for decluttering plumes are v v nice.

**Comment by PatPL** [comment]
1. Most likely 'persistentEmitters' is null in VAB. (I don't think anything else could be null directly in LateUpdate). Now I see, that I accidentally didn't copy the null checks from FixedUpdate. _Should_ be an easy fix.
2. Do the plumes disappear completely, or do they go away for a few frames? Is it related with any new settings added in 2.8.2? It would be nice if there was a way to consistently reproduce. Send logs if you manage to reproduce the bug.

**Comment by zorg2044** [comment]
2. It went away completely (including the sound) and didnt come back. Definitely not connected to the new settings as for me it happened with the old hypergolic OMS white plume in realplume not one of the new ones I was working on. 

The other user on the BDB thread had it happen multiple times including some of my custom ones (but without the declutter options as thats still in an unmerged PR) and with the original realplume solid-lower. I'll PM him and ask to submit logs here if he has any or can reproduce. I'll give it a try as well and check with different plumes.

**Comment by zorg2044** [comment]
So I've been able to reliably reproduce the issue under very specific conditions,

1. In orbit
2. A part containing realplume prefab solid-upper is fired
3. Stage using a stack separator (tested with BDB and stock seperators).
4. Fire next stage that uses a hypergolic-oms-white realplume prefab. Works fine
5. Timewarp (breaks at this point, NREs in log, plume disappears for good)

I haven't been able to contact the other user for more details but he seemed to mention other plumes that broke as well.

I've tested with a couple of other parts containing other plumes for both the first and second part but so far no "luck" with reproducing. However have been able to reliably reproduce using multiple parts that contain Solid Upper and HypergolicOMSWhite. I dont know a stock part that uses Solid Upper but I tried with a stock engine with HypergolicOMSWhite not just BDB and the issue persists.

Also for me at least its only happening if a stack separator is used in staging and not a decoupler.

I dont know how much of this is relevant or if parts of these steps are coincidental but I am able to reproduce every time. Also just to be sure I rolled back to the earlier smokescreen and it didn't happen while following the above steps.

Below logs all with fresh install of KSP 1.7.1, smokescreen 2.8.2, RealPlume+RPstock configs, BDB dev branch

Log: comparing with separator vs decoupler

https://www.dropbox.com/s/og8jidq0pfbhjvj/output_log.txt?dl=0

Log: trying few different engine combos

https://www.dropbox.com/s/cf4064g9xx5811m/output_log.txt?dl=0

Sample craft: https://www.dropbox.com/s/9g19i6fkacdespd/Test3.craft?dl=0
Requires BDB

I'll report back if I find out more.
